### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by martin-cervantes1**
:heavy_check_mark:  Added spec check for Enumerable class
:heavy_check_mark: Wrapped specs for each method in describe block.
:heavy_check_mark: Used one it block for each expected behavior. Did not test multiple aspects in one it block.
:heavy_check_mark: Tested for initial values and possible errors

### CODE REVIEW

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull request under the review by Maria Eugenia Queme P. (TSE)

### Hello @mcervantes71 and @edieatha :balloon: :lollipop: 

### Thank you for your submission, you've added test cases for all of your Enumerable methods and overall you did a good job :medal_sports: :medal_sports: there are still changes to work on :construction: :construction_worker: :construction_worker: 

#### General

- [x] Let's add an Automated Test section to your readme file and describe how to execute your test cases

#### Project specific

- [x] Specs for each method should be wrapped in one `describe` block. As for example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples . ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/describe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

you should only have one describe block per Enumerable method so instead of having this:

```ruby
 describe '#my_each' do
    it 'Check each method' do
      expect([11, 2, 3, 56].my_each { |x| p x }).to eql(4)
    end
  end

  describe '#my_each' do
    it 'Check each method' do
      expect(%w[a b c d].my_each { |x| p x }).to eql(4)
    end
  end

  describe '#my_each' do
    it 'Check each method' do
      expect([1, true, 'hi', []].my_each { |x| p x }).to eql(4)
    end
  end
```

you should have this:

```ruby
 describe '#my_each' do
    it 'Check each method' do
      expect([11, 2, 3, 56].my_each { |x| p x }).to eql(4)
    end

    it 'Check each method' do
      expect(%w[a b c d].my_each { |x| p x }).to eql(4)
    end

    it 'Check each method' do
      expect([1, true, 'hi', []].my_each { |x| p x }).to eql(4)
    end
  end
```
Change it for all your Enumerable
- [x] specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) 

You currently have no negative test cases and the positive test cases aren't descriptive please take a look at these examples and make sure to make your positive test cases more descriptive and add negative test cases for each Enumerable method

```ruby
  describe '#my_all' do
    # Positive test case with improved description
    it 'Returns true if all the elements of the Enumerable meet the given condition' do
      expect([nil, true, 99].my_any?(Integer)).to eql(true)
    end

    # Negative test cases should have negative description 
    it "Shouldn't return true if at least one of the elements of the Enumerable doesn't meet the given condition" do
      expect([1, 2].my_all?(String)).not_to eql(true)
    end

  end
```

- [x] define variables for the test cases using [`let`](https://github.com/rubocop-hq/rspec-style-guide#let-blocks). ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/before_each.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

- you use these Enumerable several times, let's put them in variables using let and use the variables instead of repeating them throughout your test cases
  - [11, 2, 3, 56]
  - %w[a b c d]
  - [1, 2]
  - (5..10)

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru**

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

**Comment by martin-cervantes1** [comment]
:heavy_check_mark: Defined global variables for the test case using let
:heavy_check_mark: Wrapped each method in one describe block
:heavy_check_mark: Added how to run test

File: README.md [hunk 1]
```diff
@@ -35,6 +35,22 @@ $ cd My_Enumerable
 require "./main.rb"
 ```
 
+### Testing
+[Pre-requisite] 
+``` sh
+gem install rspec
```
Comment (maelfosso) [inline README.md]: Create a `Gemfile` and put all the needed gems inside.
The user will just have to run `bundle install`

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello, 
Thank you for the changes you've done

1. I tweaked your tests a bit. I've got this error
```rb
Failures:

  1) Enumerable#my_each Check each method with integers array
     Failure/Error: expect(integers_array.my_each { |x| p x }).to eql(integers_array.each { |x| p x })
     
       expected: [11, 2, 3, 56, 1]
            got: 5
     
       (compared using eql?)
     # ./spec/main_spec.rb:13:in `block (3 levels) in <top (required)>'

Finished in 0.014 seconds (files took 0.09977 seconds to load)
20 examples, 1 failure

Failed examples:

rspec ./spec/main_spec.rb:12 # Enumerable#my_each Check each method with integers array
```

2. Please, instead of writing test where you print value, I suggest you compare the array result with another array or you put them into a variable and compare the result. 
It makes your tests cleaner and more understandable.

3. For `#my_all`, `#my_any`, `#my_none` please, write test using **Regex** and simple check something like `array.my_all?`

4. Read the comment left below.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by edieatha** [comment]
:heavy_check_mark: Fixed errors with some methods like each.
:heavy_check_mark: For #my_all, #my_any, #my_none tested using Regex
:heavy_check_mark: Create a Gemfile and put all the needed gems inside.
:heavy_check_mark: The user will just have to run bundle install

**Comment by uryelah** [comment]
Hello **Martin Cervantes Palomo & Edie Atha Kawanguzi**, I'm Sarah, your last reviewer for this PR 👩‍🚀  

---------    
    
Good job fixing the last required changes.

---------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**